db/version_set.cc

```diff
@@ -955,6 +970,7 @@ class LevelIterator final : public InternalIterator {
       smallest_compaction_key = (*compaction_boundaries_)[file_index_].smallest;
       largest_compaction_key = (*compaction_boundaries_)[file_index_].largest;
     }
+    CheckMayBeOutOfLowerBound();
```

Comment:
@yiwu-arbug I'm investigating an assert failure and trying to understand the code here.
Here CheckMayBeOutOfLowerBound() is called, and inside, it checks Valid() first. Valid() can be false for whatever the previous file iterator's state is. If it is false, then may_be_out_of_lower_bound_ is not updated. Did I miss anything.

Here is attached one issue I see. The assertion failure:
```
db/db_iter.cc:866: void rocksdb::DBIter::PrevInternal(): Assertion `iterate_lower_bound_ == nullptr || iter_.MayBeOutOfLowerBound() || user_comparator_.Compare(saved_key_.GetUserKey(), *iterate_lower_bound_) >= 0' failed.
```
Here is what is odd:
```
LevelIterator.may_be_out_of_lower_bound_ = false

LevelIterator.read_options_.iterate_lower_bound points to "0x00    0x00    0x00    0x00    0x04    0xbb    0x7a    0x07"

flevel_->files[file_index_].smallest_key points to "0x00    0x00    0x00    0x00    0x03    0xae    0x7f    0x99"
```
This indicates that LevelIterator.may_be_out_of_lower_bound_ is a wrong value.

Reply:
I think what you point out about the valid check make sense. How about I send a fix tomorrow and see if it fixes stress test?